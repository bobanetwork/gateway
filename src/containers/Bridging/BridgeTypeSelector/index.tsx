import { setBridgeType } from 'actions/bridgeAction'
import { setNetwork } from 'actions/networkAction'
import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import {
  selectActiveNetworkType,
  selectBridgeType,
  selectNetwork,
  selectNetworkType,
} from 'selectors'
import {
  DEFAULT_NETWORK,
  NetworkType,
  networkLimitedAvailability,
} from 'util/network/network.util'
import { BridgeTabItem, BridgeTabs } from './style'

export enum BRIDGE_TYPE {
  CLASSIC = 'CLASSIC',
  FAST = 'FAST',
  LIGHT = 'LIGHT',
  THIRD_PARTY = 'THIRD_PARTY',
}

const BridgeTypeSelector = () => {
  const dispatch = useDispatch<any>()
  const bridgeType = useSelector(selectBridgeType())
  const networkType = useSelector(selectNetworkType())
  const network = useSelector(selectNetwork())
  const isOnLimitedNetwork = networkLimitedAvailability(networkType, network)
  const activeNetworkType = useSelector(selectActiveNetworkType())

  const isMainnet =
    useSelector(selectActiveNetworkType()) === NetworkType.MAINNET

  const switchToFullySupportedNetwork = () => {
    // change network back to fully supported network when leaving light bridge
    dispatch(
      setNetwork({
        network: DEFAULT_NETWORK.chain,
        name: DEFAULT_NETWORK.name,
        networkIcon: DEFAULT_NETWORK.icon,
        chainIds: DEFAULT_NETWORK.chainId,
        networkType,
      })
    )
  }

  useEffect(() => {
    if (bridgeType !== BRIDGE_TYPE.LIGHT && isOnLimitedNetwork) {
      switchToFullySupportedNetwork()
    }
  }, [bridgeType, isOnLimitedNetwork])

  const onTabClick = (payload: any) => {
    dispatch(setBridgeType(payload))
  }

  useEffect(() => {
    dispatch(setBridgeType(BRIDGE_TYPE.CLASSIC))
  }, [activeNetworkType])

  return (
    <BridgeTabs>
      <BridgeTabItem
        data-testid="classic-btn"
        active={bridgeType === BRIDGE_TYPE.CLASSIC}
        onClick={() => onTabClick(BRIDGE_TYPE.CLASSIC)}
      >
        Classic
      </BridgeTabItem>

      <BridgeTabItem
        data-testid="light-btn"
        active={bridgeType === BRIDGE_TYPE.LIGHT}
        onClick={() => onTabClick(BRIDGE_TYPE.LIGHT)}
      >
        Light
      </BridgeTabItem>

      {isMainnet ? (
        <BridgeTabItem
          data-testid="third-party-btn"
          active={bridgeType === BRIDGE_TYPE.THIRD_PARTY}
          onClick={() => onTabClick(BRIDGE_TYPE.THIRD_PARTY)}
        >
          Third Party
        </BridgeTabItem>
      ) : null}
    </BridgeTabs>
  )
}

export default BridgeTypeSelector
