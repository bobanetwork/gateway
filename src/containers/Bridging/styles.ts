import { Button, Typography } from 'components/global'
import styled, { css } from 'styled-components'
import { mobile } from 'themes/screens'

export const BridginContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 32px;
  max-width: 500px;
  margin: 32px auto;

  ${mobile(css`
    max-width: 400px;
    margin: 24px auto;
  `)}
`

export const BridgeWrapper = styled.div`
  padding: 24px;
  display: flex;
  flex-direction: column;
  gap: 8px;
  width: 100%;
  border-radius: 12px;
  border: 1px solid
    ${({ theme: { colors, name } }) =>
      name === 'light' ? colors.gray[400] : colors.gray[300]};
  background: ${({ theme }) => theme.colors.box.background};
  /* Gradient Glass BG Blur */
  backdrop-filter: blur(7.5px);

  ${mobile(css`
    padding: 24px 16px;
  `)}
`

export const BridgeContent = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 20px;
  justify-content: space-around;
  align-items: flex-start;
`
export const BridgeActionContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: around;
  align-items: center;
  margin-top: 24px;
`

export const BridgeActionButton = styled(Button).attrs({
  style: {
    width: '100%',
  },
})``

export const Label = styled(Typography).attrs({
  variant: 'body3',
})`
  font-weight: 400;
  line-height: normal;
  color: ${({ theme, color }) =>
    color
      ? color
      : theme.name === 'light'
        ? theme.colors.gray[700]
        : theme.colors.gray[100]};
`
