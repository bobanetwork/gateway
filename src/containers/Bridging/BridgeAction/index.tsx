import { setConnect } from 'actions/setupAction'
import { openModal } from 'actions/uiAction'
import { Heading } from 'components/global'
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import {
  selectAccountEnabled,
  selectActiveNetworkType,
  selectAmountToBridge,
  selectBridgeAlerts,
  selectTokenToBridge,
} from 'selectors'
import { BridgeActionButton, BridgeActionContainer } from '../styles'
import { ONRAMP_API_KEY, PURCHASE_RAMP_URL } from 'util/constant'
import { NetworkType } from 'util/network/network.util'
import { trackClick } from 'util/analytics'

const BridgeAction = () => {
  const dispatch = useDispatch<any>()
  const accountEnabled = useSelector<any>(selectAccountEnabled())
  const token = useSelector(selectTokenToBridge())
  const amountToBridge = useSelector(selectAmountToBridge())
  const bridgeAlerts = useSelector(selectBridgeAlerts())

  const isMainnet =
    useSelector(selectActiveNetworkType()) === NetworkType.MAINNET

  const isBridgeActionDisabled = () => {
    const hasError = bridgeAlerts.find((alert: any) => alert.type === 'error')
    return !token || !amountToBridge || hasError
  }

  const onConnect = () => {
    dispatch(setConnect(true))
  }

  const onBridge = () => {
    if (!isBridgeActionDisabled()) {
      dispatch(openModal({ modal: 'bridgeConfirmModal' }))
    }
  }

  return (
    <BridgeActionContainer>
      {!accountEnabled ? (
        <BridgeActionButton
          onClick={onConnect}
          data-testid="connect-btn"
          label={<Heading variant="h3">Connect Wallet</Heading>}
        />
      ) : (
        <div
          style={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            justifyContent: 'center',
            width: '100%',
            gap: '10px',
          }}
        >
          <BridgeActionButton
            disabled={isBridgeActionDisabled()}
            onClick={onBridge}
            data-testid="bridge-btn"
            label={<Heading variant="h3">Bridge</Heading>}
          />
          {isMainnet && ONRAMP_API_KEY ? (
            <BridgeActionButton
              outline
              onClick={() => {
                trackClick('button', 'button_click', 'Buy With Ramp Network')
                window.open(PURCHASE_RAMP_URL, 'blank')
              }}
              data-testid="connect-btn"
              label={<Heading variant="h3">Buy with Ramp Network</Heading>}
            />
          ) : null}
        </div>
      )}
    </BridgeActionContainer>
  )
}

export default BridgeAction
