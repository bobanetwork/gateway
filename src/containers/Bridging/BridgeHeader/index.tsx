import HelpOutlineOutlined from '@mui/icons-material/HelpOutlineOutlined'
import { openModal } from 'actions/uiAction'
import { Heading } from 'components/global'
import Tooltip from 'components/tooltip/Tooltip'
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { selectActiveNetworkType } from 'selectors'
import styled, { useTheme } from 'styled-components'
import { NetworkType } from 'util/network/network.util'
import { BridgeHeaderWrapper, GearIcon, IconWrapper } from './styles'
import { useNetworkInfo } from 'hooks/useNetworkInfo'

export const LabelStyle = styled.span`
  color: var(--Gray-50, #eee);
  font-family: Roboto;
  font-size: 12px;
  font-style: normal;
  font-weight: 500;
  line-height: 138.3%;
`

export const ValueStyle = styled.span`
  color: var(--Gray-50, #eee);
  font-family: Roboto;
  font-size: 12px;
  font-style: normal;
  font-weight: 400;
  line-height: 138.3%;
`

const BridgeHeader = () => {
  const dispatch = useDispatch<any>()
  const theme: any = useTheme()
  const { isSepoliaNetwork } = useNetworkInfo()
  const isTestnet =
    useSelector(selectActiveNetworkType()) === NetworkType.TESTNET

  const iconColor =
    theme.name === 'light' ? theme.colors.gray[600] : theme.colors.gray[100]

  const openSettingModal = () => {
    dispatch(openModal('settingsModal'))
  }

  const ClassicBridgeInfo = () => {
    return (
      <>
        <LabelStyle>Classic Bridge</LabelStyle> <br />
        <ValueStyle>
          Although this option is always available, it takes 7 days to receive
          your funds when withdrawing from L2 to L1.
        </ValueStyle>
        <br />
        <br />
      </>
    )
  }

  const FastBridgeInfo = () => {
    return (
      <>
        <LabelStyle>Fast Bridge</LabelStyle>
        <br />
        <ValueStyle>
          A swap-based bridge to Boba L2. This option is only available if the
          pool balance is sufficient.
        </ValueStyle>
        <br />
        <br />
      </>
    )
  }

  const LightBridgeInfo = () => {
    if (!isTestnet) {
      return <></>
    }
    return (
      <>
        {' '}
        <LabelStyle>Light Bridge</LabelStyle>
        <br />
        <ValueStyle>
          Bridge assets instantaneously and even between L2's. This option is
          only available for a few selected assets (mostly BOBA).
        </ValueStyle>
      </>
    )
  }

  return (
    <BridgeHeaderWrapper>
      <Heading variant="h2">
        Bridge
        {isSepoliaNetwork ? null : (
          <Tooltip
            data-testid="tooltip-btn"
            title={
              <>
                <ClassicBridgeInfo />
                <FastBridgeInfo />
                <LightBridgeInfo />
              </>
            }
          >
            <IconWrapper inline={true} style={{ marginLeft: '5px' }}>
              <HelpOutlineOutlined
                fontSize="small"
                sx={{ cursor: 'pointer', color: iconColor }}
              />
            </IconWrapper>
          </Tooltip>
        )}
      </Heading>
      <IconWrapper>
        <GearIcon
          sx={{ color: iconColor }}
          onClick={openSettingModal}
          data-testid="setting-btn"
        />
      </IconWrapper>
    </BridgeHeaderWrapper>
  )
}

export default BridgeHeader
