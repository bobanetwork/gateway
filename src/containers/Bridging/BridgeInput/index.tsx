import { openModal } from 'actions/uiAction'
import React, { FC } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { selectAccountEnabled, selectTokenToBridge } from 'selectors'

import useBridgeSetup from 'hooks/useBridgeSetup/'
import { getCoinImage } from 'util/gitdata'

import bobaLogo from 'assets/images/Boba_Logo_White_Circle.png'
import useAmountToReceive from 'hooks/useAmountToReceive'
import { useNetworkInfo } from 'hooks/useNetworkInfo'
import { SectionLabel } from '../chain/styles'
import BridgeToAddress from './BridgeToAddress'
import EmergencySwap from './EmergencySwap'
import Fee from './Fee'
import TokenInput from './TokenInput'
import {
  BridgeInputContainer,
  BridgeInputWrapper,
  DownArrow,
  ReceiveAmount,
  ReceiveContainer,
  TokenLabel,
  TokenPickerIcon,
  TokenSelector,
  TokenSelectorInput,
  TokenSymbol,
} from './styles'

const BridgeInput: FC = () => {
  const dispatch = useDispatch<any>()
  const isAccountEnabled = useSelector(selectAccountEnabled())
  const token = useSelector(selectTokenToBridge())
  const { amount: receivableAmount } = useAmountToReceive()
  const { isActiveNetworkBnb } = useNetworkInfo()
  useBridgeSetup()

  const openTokenPicker = () => {
    dispatch(openModal({ modal: 'tokenPicker' }))
  }

  if (!isAccountEnabled) {
    return null
  }

  return (
    <BridgeInputContainer>
      <BridgeInputWrapper>
        <TokenSelector>
          <SectionLabel>Token</SectionLabel>
          <TokenSelectorInput
            onClick={() => openTokenPicker()}
            id="tokenSelectorInput"
          >
            {token && (
              <TokenSymbol>
                <img
                  src={
                    token.symbol === 'BOBA'
                      ? bobaLogo
                      : getCoinImage(token.symbol)
                  }
                  alt={`ETH logo`}
                  width="32px"
                  height="32px"
                />
              </TokenSymbol>
            )}
            <TokenLabel>{token ? token?.symbol : 'Select'}</TokenLabel>
            <TokenPickerIcon>
              <DownArrow />
            </TokenPickerIcon>
          </TokenSelectorInput>
        </TokenSelector>
        <TokenInput />
      </BridgeInputWrapper>
      {token && (
        <ReceiveContainer>
          <SectionLabel>Receive</SectionLabel>
          <ReceiveAmount data-testid="amountToRecieve">
            {receivableAmount}
          </ReceiveAmount>
        </ReceiveContainer>
      )}
      <BridgeToAddress />
      {token && <Fee />}
      {!!isActiveNetworkBnb && <EmergencySwap />}
    </BridgeInputContainer>
  )
}

export default BridgeInput
