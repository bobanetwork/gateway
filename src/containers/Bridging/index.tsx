import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { selectBridgeType } from 'selectors'
import BridgeAction from './BridgeAction'
import BridgeAlert from './BridgeAlert'
import BridgeHeader from './BridgeHeader'
import BridgeInput from './BridgeInput'
import BridgeTypeSelector, { BRIDGE_TYPE } from './BridgeTypeSelector'
import ThirdPartyBridges from './ThirdPartyBridges'
import Chains from './chain'
import { BridgeContent, BridgeWrapper, BridginContainer } from './styles'

import useBridgeAlerts from 'hooks/useBridgeAlerts'
import useBridgeCleanUp from 'hooks/useBridgeCleanUp'
import ReenterWithdrawModal from '../modals/ReenterWithdrawModal'
import networkService from '../../services/networkService'
import { openModal } from '../../actions/uiAction'
import { setReenterWithdrawalConfig } from '../../actions/bridgeAction'
import { bedrockGraphQLService } from '../../services/graphql.service'
import {
  ReenterWithdrawConfig,
  WithdrawState,
} from '../modals/MultiStepWithdrawalModal/withdrawal'

const Bridging = () => {
  const dispatch = useDispatch<any>()
  useBridgeCleanUp()
  useBridgeAlerts()

  const [reenterWithdrawConfig, setReenterWithdrawConfig] =
    useState<ReenterWithdrawConfig | null>()

  useEffect(() => {
    networkService.provider?.getNetwork().then(async (res) => {
      bedrockGraphQLService
        .queryWithdrawalTransactionsHistory(
          await networkService.provider!.getSigner().getAddress(),
          // TODO fetch network config
          {
            L2: {
              name: 'ETH',
              chainId: 901,
            },
          } as any
        )
        .then((events: any) => {
          events = events.filter(
            (e) => e.UserFacingStatus !== WithdrawState.finalized
          )
          if (events?.length > 1) {
            if (events[0]?.actionRequired) {
              setReenterWithdrawConfig({
                ...events[0].actionRequired,
              })
            }
          }
        })
    })
  }, [!!networkService.provider])

  const currentBridgeType = useSelector(selectBridgeType())

  const handleClose = () => setReenterWithdrawConfig(null)

  return (
    <BridginContainer>
      <BridgeWrapper id="bridge">
        {reenterWithdrawConfig && (
          <ReenterWithdrawModal
            handleClose={handleClose}
            state={1}
            onReenterWithdrawal={() => {
              dispatch(setReenterWithdrawalConfig(reenterWithdrawConfig))
              dispatch(openModal('bridgeMultiStepWithdrawal'))
            }}
          />
        )}
        <BridgeContent>
          <BridgeHeader />
          <BridgeAlert />
          <BridgeTypeSelector />
          {currentBridgeType !== BRIDGE_TYPE.THIRD_PARTY ? (
            <>
              <Chains />
              <BridgeInput />
            </>
          ) : (
            <ThirdPartyBridges />
          )}
        </BridgeContent>
        {currentBridgeType !== BRIDGE_TYPE.THIRD_PARTY ? (
          <BridgeAction />
        ) : null}
      </BridgeWrapper>
    </BridginContainer>
  )
}

export default Bridging
