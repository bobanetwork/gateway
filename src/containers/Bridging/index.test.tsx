import { render } from '@testing-library/react'
import useBridgeAlerts from 'hooks/useBridgeAlerts'
import useBridgeCleanup from 'hooks/useBridgeCleanup'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import CustomThemeProvider from 'themes'
import { NETWORK, NETWORK_TYPE } from 'util/network/network.util'
import Bridging from '.'
import { BRIDGE_TYPE } from './BridgeTypeSelector'
import { mockedInitialState } from 'util/tests'

const mockStore = configureStore([thunk])

jest.mock('hooks/useBridgeAlerts')
jest.mock('hooks/useBridgeCleanup')

const mockUseBridgeAlerts = useBridgeAlerts as jest.MockedFunction<
  typeof useBridgeAlerts
>
const mockUseBridgeCleanup = useBridgeCleanup as jest.MockedFunction<
  typeof useBridgeCleanup
>

const renderBridging = ({ store }: any) => {
  return render(
    <Provider store={store}>
      <CustomThemeProvider>
        <Bridging />
      </CustomThemeProvider>
    </Provider>
  )
}

describe('Bridging Component', () => {
  let store: any

  beforeEach(() => {
    store = mockStore(mockedInitialState)
  })

  test('should match snapshot on when bridge type is CLASSIC', () => {
    const { asFragment } = renderBridging({ store })
    expect(asFragment()).toMatchSnapshot()
    expect(mockUseBridgeAlerts).toHaveBeenCalled()
    expect(mockUseBridgeCleanup).toHaveBeenCalled()
  })

  test('should match snapshot on when bridge type is THIRD_PARTY', () => {
    store = mockStore({
      ...mockedInitialState,
      bridge: {
        ...mockedInitialState.bridge,
        bridgeType: BRIDGE_TYPE.THIRD_PARTY,
      },
    })

    const { asFragment } = renderBridging({ store })
    expect(asFragment()).toMatchSnapshot()
  })
})
