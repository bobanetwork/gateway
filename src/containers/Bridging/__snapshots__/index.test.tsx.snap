// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bridging Component should match snapshot on when bridge type is CLASSIC 1`] = `
<DocumentFragment>
  <div
    class="sc-eldPxv dcLceF"
  >
    <div
      class="sc-fPXMVe dUoduF"
      id="bridge"
    >
      <div
        class="sc-gFqAkR bGXBoY"
      >
        <div
          class="sc-koXPp ejMtSJ"
        >
          <h2
            class="sc-eDPEul jXgUQt"
          >
            Bridge
            <div
              class="sc-bmzYkS dMUwNo"
              data-mui-internal-clone-element="true"
              data-testid="tooltip-btn"
              style="margin-left: 5px;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1t96nry-MuiSvgIcon-root"
                data-testid="HelpOutlineOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
                />
              </svg>
            </div>
          </h2>
          <div
            class="sc-bmzYkS JFHbA"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium sc-iHGNWf fwjtTZ css-1g4na4e-MuiSvgIcon-root"
              data-testid="setting-btn"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4m0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-dtInlm hzapnV"
        >
          <div
            class="sc-kOPcWz gfXFER"
            data-testid="classic-btn"
          >
            Classic
          </div>
          <div
            class="sc-kOPcWz hxKNaX"
            data-testid="light-btn"
          >
            Light
          </div>
          <div
            class="sc-kOPcWz hxKNaX"
            data-testid="third-party-btn"
          >
            Third Party
          </div>
        </div>
        <div
          class="sc-cfxfcM icLxWe"
        >
          <div
            class="sc-gFAWRd jXlJGV"
          >
            <p
              class="sc-imWYAI dZiXLh sc-hIUJlX fFfgEP"
            >
              From
            </p>
            <div
              class="sc-hRJfrW gQTHzT"
            >
              <div
                class="sc-iHbSHJ cMMGQH"
              >
                <svg
                  fill="none"
                  height="32"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="#5B78ED"
                    height="32"
                    rx="16"
                    width="32"
                  />
                  <path
                    d="M16.147 5.33333L16.0039 5.81949V19.9256L16.147 20.0684L22.6948 16.1979L16.147 5.33333Z"
                    fill="#E5E7EB"
                  />
                  <path
                    d="M16.148 5.33333L9.6001 16.1979L16.148 20.0684V13.2217V5.33333Z"
                    fill="#F9F9FA"
                  />
                  <path
                    d="M16.1471 21.308L16.0664 21.4064V26.4312L16.1471 26.6667L22.6988 17.4396L16.1471 21.308Z"
                    fill="#E5E7EB"
                  />
                  <path
                    d="M16.148 26.6667V21.308L9.6001 17.4396L16.148 26.6667Z"
                    fill="#F9F9FA"
                  />
                  <path
                    d="M16.147 20.0681L22.6947 16.1977L16.147 13.2214V20.0681Z"
                    fill="#D2D5DA"
                  />
                  <path
                    d="M9.6001 16.1977L16.148 20.0681V13.2214L9.6001 16.1977Z"
                    fill="#E5E7EB"
                  />
                </svg>
              </div>
              <p
                class="sc-imWYAI hMxxJk sc-klVQfs eaiVHy"
              >
                ethereum
              </p>
              <div
                class="sc-bypJrT jpBGQA"
              >
                <div
                  class="sc-ddjGPC ipgvoq"
                >
                  <div>
                    <div
                      data-src="arrowdown.svg"
                      src=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-dSCufp dvHzRa"
            id="switchBridgeDirection"
          >
            <div
              class="sc-fxwrCY beyRDV"
            >
              <div>
                <div
                  data-src="switchIcon.svg"
                  src=""
                />
              </div>
            </div>
          </div>
          <div
            class="sc-gFAWRd jXlJGV"
          >
            <p
              class="sc-imWYAI dZiXLh sc-hIUJlX fFfgEP"
            >
              To
            </p>
            <div
              class="sc-hRJfrW gQTHzT"
            >
              <div
                class="sc-iHbSHJ cMMGQH"
              >
                <svg
                  fill="none"
                  viewBox="0 0 1202 1202"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="601"
                    cy="601"
                    fill="white"
                    r="601"
                  />
                  <path
                    d="M682 711.3H375.9V264H682C787.1 264 872.2 349.2 872.2 454.2V521C872.2 626.1 787.1 711.3 682 711.3Z"
                    fill="#ACD905"
                  />
                  <path
                    d="M760.9 935H375.9V711.3C375.9 587.8 476 487.7 599.5 487.7H760.9C884.4 487.7 984.5 587.8 984.5 711.3C984.5 834.8 884.4 935 760.9 935Z"
                    fill="#AEAFAA"
                  />
                  <path
                    d="M872.2 517.4V521.1C872.2 626.2 787.1 711.3 681.9 711.3H375.9C375.9 649.6 401 593.7 441.4 553.2C481.9 512.7 537.8 487.7 599.6 487.7H760.9C801.5 487.7 839.5 498.5 872.2 517.4Z"
                    fill="#5D6E3C"
                  />
                  <path
                    d="M376.5 824C500.488 824 601 723.488 601 599.5C601 475.512 500.488 375 376.5 375C252.512 375 152 475.512 152 599.5C152 723.488 252.512 824 376.5 824Z"
                    fill="#627EEA"
                  />
                  <path
                    d="M376.5 403.2L373.9 412.2V671.7L376.5 674.4L497 603.1L376.5 403.2Z"
                    fill="#C5C5C5"
                  />
                  <path
                    d="M376.5 403.2L256 603.1L376.5 674.4V548.4V403.2Z"
                    fill="#E7E7E7"
                  />
                  <path
                    d="M376.5 697.2L375 699V791.4L376.5 795.8L497.1 626L376.5 697.2Z"
                    fill="#C5C5C5"
                  />
                  <path
                    d="M376.5 795.8V697.2L256 626L376.5 795.8Z"
                    fill="#E7E7E7"
                  />
                  <path
                    d="M376.5 674.4L497 603.1L376.5 548.4V674.4Z"
                    fill="#909090"
                  />
                  <path
                    d="M256 603.1L376.5 674.4V548.4L256 603.1Z"
                    fill="#AEAEAE"
                  />
                </svg>
              </div>
              <p
                class="sc-imWYAI hMxxJk sc-klVQfs eaiVHy"
              >
                boba
              </p>
              <div
                class="sc-bypJrT jpBGQA"
              >
                <div
                  class="sc-ddjGPC ipgvoq"
                >
                  <div>
                    <div
                      data-src="arrowdown.svg"
                      src=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-feUZmu fKaVOM"
      >
        <button
          class="sc-jlZhew fFYrTq sc-fUnMCh"
          data-testid="connect-btn"
          label="[object Object]"
          style="width: 100%;"
          type="button"
        >
           
          <h3
            class="sc-eDPEul CjCmp"
          >
            Connect Wallet
          </h3>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Bridging Component should match snapshot on when bridge type is THIRD_PARTY 1`] = `
<DocumentFragment>
  <div
    class="sc-eldPxv dcLceF"
  >
    <div
      class="sc-fPXMVe dUoduF"
      id="bridge"
    >
      <div
        class="sc-gFqAkR bGXBoY"
      >
        <div
          class="sc-koXPp ejMtSJ"
        >
          <h2
            class="sc-eDPEul jXgUQt"
          >
            Bridge
            <div
              class="sc-bmzYkS dMUwNo"
              data-mui-internal-clone-element="true"
              data-testid="tooltip-btn"
              style="margin-left: 5px;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1t96nry-MuiSvgIcon-root"
                data-testid="HelpOutlineOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
                />
              </svg>
            </div>
          </h2>
          <div
            class="sc-bmzYkS JFHbA"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium sc-iHGNWf fwjtTZ css-1g4na4e-MuiSvgIcon-root"
              data-testid="setting-btn"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4m0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-dtInlm hzapnV"
        >
          <div
            class="sc-kOPcWz hxKNaX"
            data-testid="classic-btn"
          >
            Classic
          </div>
          <div
            class="sc-kOPcWz hxKNaX"
            data-testid="light-btn"
          >
            Light
          </div>
          <div
            class="sc-kOPcWz gfXFER"
            data-testid="third-party-btn"
          >
            Third Party
          </div>
        </div>
        <div
          class="sc-gFVvzn cJRSzO"
        >
          <p
            class="sc-imWYAI dZiXLh sc-hIUJlX fFfgEP"
          >
            Third party bridges
          </p>
          <div
            class="sc-brPLxw cRoaAp"
          >
            <p
              class="sc-imWYAI hMxxJk"
            >
              Loading ...
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
